graphqlUtils_test.go

```diff
@@ -0,0 +1,353 @@
+package resource
+
+import (
+	"regexp"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func assertEqualNoWhitespace(t *testing.T, expected, actual string) {
+	whitespace := regexp.MustCompile(`\s+`)
+	assert.Equal(t, strings.Trim(whitespace.ReplaceAllString(expected, " "), " "), strings.Trim(whitespace.ReplaceAllString(actual, " "), " "))
+}
+
+func TestSingleFieldInsertIntoQueryType(t *testing.T) {
+	combined, err := combineSchemas(`
+	type Query {
+	}`, `
+	type Query {
+		hello: String
+	}`, []Resolver{
+		{
+			FieldName: "hello",
+			TypeName:  "Query",
+		},
+	})
+
+	assert.Nil(t, err)
+	assertEqualNoWhitespace(t, `type Query { hello: String }`, combined)
+}
+
+func TestSingleFieldNotInResolverSet(t *testing.T) {
+	combined, err := combineSchemas(`
+	type Query {
+	}`, `
+	type Query {
+		hello: String
+	}`, []Resolver{})
+
+	assert.Nil(t, err)
+	assertEqualNoWhitespace(t, `type Query {}`, combined)
+}
```

Comment:
This will create a mismatch between a local GraphQL schema and remote GraphQL schema

Reply:
it will, in the same way as there's gonna be a mismatch between local function code and remote deployed function code.

Reply:
But I think this resource/action should be separated from the structure of the consuming project ie.ch-api.

Reply:
The structure is already dictated by `resolvers_file`. It is completely OK to invoke this action with different resolver files to achieve partial updates. This just ensures that when you do partial updates the schema will be consistent with the resolvers.

Reply:
I tried to clear my point a bit more regarding the schema mismatch in the next comment 👇